app/src/main/kotlin/io/intrepid/skotlinton/base/BaseFragment.kt

```diff
@@ -11,18 +14,34 @@ import android.view.ViewGroup
 import butterknife.ButterKnife
 import butterknife.Unbinder
 import io.intrepid.skotlinton.SkotlintonApplication
+import io.reactivex.disposables.CompositeDisposable
 import timber.log.Timber
 
-abstract class BaseFragment<out P : BaseContract.Presenter> : Fragment(), BaseContract.View {
+abstract class BaseFragment<out VM : BaseViewModel> : Fragment() {
 
     protected val skotlintonApplication: SkotlintonApplication
-        get() = activity!!.application as SkotlintonApplication
+        get() = requireActivity().application as SkotlintonApplication
     protected abstract val layoutResourceId: Int
 
-    protected val presenter: P by lazy(LazyThreadSafetyMode.NONE) {
-        val configuration = skotlintonApplication.getPresenterConfiguration()
-        createPresenter(configuration)
+    protected val onPauseDisposable = CompositeDisposable()
+    protected val onStopDisposable = CompositeDisposable()
+    protected val onDestroyViewDisposable = CompositeDisposable()
+    protected val onDestroyDisposable = CompositeDisposable()
```

Comment:
I still don't love having all of these but I think it works as a first pass and we can think further about it later.

Reply:
This seems like a good argument for choosing to base our MVVM off of LiveData instead of Rx

Reply:
I am leaning towards to migrating to LiveData as well, mainly due to the fact that it knows to not retrigger the observer when resuming if the value hasn't changed. We probably can't entirely avoid doing something like this though, most projects I worked on inevitably end up needing to override multiple lifecycle events.